### ISSUE TITLE
Canon EOS R5 Raw image problems

### ISSUE THREAD

**Issue by rebio** [Beep6581/RawTherapee#6438]
**Short description**
When I open a raw image from my new EOS R5 with RT pretty much every image looks incredibly bad and needs a ton of tweaking no get it somewhat right.

**Problem in detail**
Pretty much all images have one or more of the following problems:
- Whitebalance is off a huge amount (could be related to #6089 )
- Black border on two sides of the image (eg. left and top). On the JPG version of the same file there is no such border at all.
- Red Gradient from one side (eg. left side). On the JPG version of the same file there is no such gradient at all.
- Terrible looking curve preset (eg. way too bright, no contrast, etc.)

I am not able to reproduce one problem in an image when I take another image (so I have no idea what causes the issue). But the image looks always the same in RT every time when I open it (so I know that it's not a one-hit-bug that is never to be seen ever again)

As said: These problems can be fixed by single tweaking pretty much every setting you have in RT, but it looks more like that an incorrect preset is chosen by default.

**Comparison with Lightroom**
I only had the time to check the first problem (white balance) in Lightroom. When I import my images into Lightroom they look OK (or at least as good as I would expect them to look).

**Example images**
| Example      | RT out (1) |  EOS out (2) | RT custom (3) | LR out (4) |
| :-----------: | :-----------: | :-----------: | :-----------: | :-----------: |
| Image 1 A   | ![01a RT std out](https://user-images.githubusercontent.com/10911084/159160631-7e6682b0-fbd9-4345-8a38-6acddf6b957c.jpg)  | ![01b EOS std out](https://user-images.githubusercontent.com/10911084/159160636-6bbfc99e-bc48-4804-bf3c-5311d869bee3.jpg) | ![01c RT custom out](https://user-images.githubusercontent.com/10911084/159160640-5c0e4431-9ca6-4e8a-bc9e-32754d20771a.jpg) | ![01d LR std out](https://user-images.githubusercontent.com/10911084/159160641-8919132e-5fe5-43f8-8fbd-6f59b20fcd23.jpg) |
| Image 1 B   | ![01aa RT std out](https://user-images.githubusercontent.com/10911084/159160664-f61fe978-2548-46cf-8d59-f8ca7b3764ef.jpg) | ![01ba EOS std out](https://user-images.githubusercontent.com/10911084/159160670-45e2ec6a-d8e3-4da2-ba79-b253f648a70f.jpg) | ![01ca RT custom out](https://user-images.githubusercontent.com/10911084/159160672-ca6e2c42-0665-4173-b828-b161712c2c1b.jpg)| ![01da LR std out](https://user-images.githubusercontent.com/10911084/159160676-e170222e-f615-4f7c-94d3-680d2b395d50.jpg) |
| Image 2   | ![02a RT std out](https://user-images.githubusercontent.com/10911084/159160769-95ca2137-624f-4053-a23b-05c406fddb52.jpg) | ![02b EOS std out](https://user-images.githubusercontent.com/10911084/159160774-838e16aa-9650-451a-8326-6b05452efbeb.jpg) | / | coming soon (TM) <br/>In short: There are no black borders |
| Image 3   | ![03a RT std out](https://user-images.githubusercontent.com/10911084/159160787-bb90837c-71cd-4243-8421-d5bbb658f77d.jpg) | ![03b EOS std out](https://user-images.githubusercontent.com/10911084/159160788-ebf7dd61-46da-45bc-bea7-77f09c648437.jpg) | / | ![03d LR std out](https://user-images.githubusercontent.com/10911084/159164984-3c59b188-8543-44e1-a8e6-7cd22b40612a.jpg) |
| Image 4 | ![04a RT std out](https://user-images.githubusercontent.com/10911084/159162168-5df0d59f-3d57-46ae-9236-ab453e1eca36.jpg) | ![04b2 EOS std out](https://user-images.githubusercontent.com/10911084/159164917-8731159f-ebcb-4c6d-a5bb-789297bcb337.jpg) | / | ![04d2 LR std out](https://user-images.githubusercontent.com/10911084/159164902-6d67d245-ba41-4b15-a645-4ea52f61dc42.jpg) |

1) RT out: RawTherapee with the default profile (_as opened_)
2) EOS out: Image as it is saved to JPG by the camera
3) RT custom: Processed with RT
4) LR out: Lightroom with the default profile (_as opened_) aka 1) for Lightroom

- Image 1) White balance is terribly off. A is just some part. B is an eye. Note the magenta highlight in the eye that actually is a lamp (in RT custom). Not sure what caused this, but I guess it was that I tried to fix the issue with the wrong tools.
- Image 2) Many images have this black border on two sides. No idea why. (Screenshot of the image in GIMP)
- Image 3) Many images have this red/magenta Gradient coming in from one side. (Screenshot of the image in GIMP)
- Image 4) Some images look like they are using a very bad preset. Because of testing I would say they use the _Auto-Matched Curve_. The Standard Film Curve looks pretty much as bad in the other direction. So auto matched is way too bright and no contrast and standard film is the same thing, but with dark instead of bright.

I recently updated the camera Firmware from 1.4 to 1.5. Since I have not seen these problems in RT with the same camera + lens before, this might (?) be a potential issue.

**Steps to reproduce**
1. Open image with RT
2. Export what you see

**Expected behavior**
The image should look somewhat OK. Of course you have to tweak here and there, but it shouldn't be that bad.

**Additional information**
Info from _AboutThisBuild.txt_
- Version: 5.8-3075-g5502f52f9
- Branch: dev
- Commit: 5502f52f9
- Commit date: 2022-02-21
- Compiler: cc 11.2.0
- Processor: generic x86
- System: Windows
- Bit depth: 64 bits
- Gtkmm: V3.24.5
- Lensfun: V0.3.2.0
- Build type: release
- Build flags:  -std=c++11 -fno-tree-loop-vectorize -mtune=generic -Werror=unused-label -Werror=delete-incomplete -fno-math-errno -Wno-attributes -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -Wunused-macros -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
- Link flags:  -mtune=generic
- OpenMP support: ON
- MMAP support: ON
- Build OS: MINGW64_NT-10.0-20348 3.3.4-341.x86_64 x86_64
- Build date: Mon, 21 Feb 2022 10:08:56 +0000 UTC
- Build epoch: 1645438136
- Build UUID: e7d348b3-c00c-4f38-83a7-e50ed8367d97

Other useful information:
I tested around a bit and also checked what would happen when I transform the CR3 to DNG and process the DNG in RT. I converted the CR3 file using this online tool: https://www.onlineconverter.com/cr3-to-dng
Long story short: It's also bad, but not nearly as bad. There are no black borders, there is no red gradient. The light and the colours still look way too dark, but not nearly as bad as the default CR3 file.

As additional use: I used the camera with and RF to EF adapter with an EF mount lens. I also used an EF-S lens on the same setup. For the EF-S lens the camera switched into a 1,6 crop mode, but it still works like every other lens by the looks of it.

I will try to add the images. IIRC github does not allow ~30-50MB raw files to be uploaded directly.
I'll try to do my best

**Comment by rebio** [comment]
More testing:
1)
- The black border seems to be linked to the 1.6x crop mode. I get the exact same black border on two sides with the read gradient when I use test imaged from https://raw.pixls.us/. I have tested all files from the page and I got these results.

| Remark | Dual Pixel Raw | 1,6crop | CRAW | comment | Image checksum |
| :-----: | :-----: | :-----: | :-----: | :-----: | :-----: |
| _none_ | yes | yes | no | listed in RT,<br/>has black border | d5f66b48011d84308b0512d769d9f5b5e2a84e59 |
| RAW dualpixel | yes | no | no | listed in RT,<br/>has **no** black border | f8a5194c1ed2a79f57e17ca385862e1b8e75eeef |
| CRAW 1.6crop dualpixel | yes | yes | yes | **not** listed in RT | e6cfcede65d97e3a50b55d4c44728deb7f817d82 |
| RAW 1.6crop no dualpixel | no | yes | no | listed in RT,<br/>has black border | dfbd245fa878fff2a79baf0db4f258d9a7798339 |
| CRAW 1.6crop no dualpixel | no | yes | yes | **not** listed in RT | a704fe3a62fbcd4aac431394e41d32307f1ba70f |
| CRAW dualpixel | yes | no | yes | **not** listed in RT | 5b9704c191d4ec5ec078d81f6c123208095181a4 |
| RAW no dualpixel | no | no | no | listed in RT,<br/>has **no** black border | 521122ad294e23ebdaf91e8b34ded9beb5c3348f |
| CRAW no dualpixel | no | no | yes | **not** listed in RT | 1065866a5cd1c88f73f36b534508af4aff534b5e |

- Seems like CRAW does not work at all
- Seems like the black border only appears in 1.6 crop mode
- I did only had OK looking default settings for these images

2)
| Image | Preset _Auto Matched_ | Preset _Standard Film Curve_ |
| :-----: | :-----: | :-----: |
| Own Image |![Auto matched ISO Low](https://user-images.githubusercontent.com/10911084/159548520-a0543669-d315-4167-a392-c59ce9864c72.png) |![Standard film curve ISO Low](https://user-images.githubusercontent.com/10911084/159548533-78dd6dac-8c40-4dfe-a1f2-939b2093bb81.png) |
| Sample Image |![Auto matched ISO Low sample](https://user-images.githubusercontent.com/10911084/159548569-cfc37673-8c65-4d6b-afcf-54760ec98498.png) |![Standard film curve ISO Low sample](https://user-images.githubusercontent.com/10911084/159548598-2a663326-bb87-49ee-ad97-6795a798dc28.png) |

The auto matched preset (that usually is pretty good fit) seems to be incredibly off. This problem only seems to exist for some images, but not all. Many of my shots before did not have this problem. I used an APSC lens (EF-S mount) and switched back to a full frame lens (EF mount). I did not know that the camera would not switch back to full frame so it stayed in the 1,6 crop mode. Maybe this is related?

**Comment by Thanatomanic** [comment]
Hi @rebio Thanks for the extensive write-up. A quick reply to your very last point about CRAW. This should be fixed soon: https://github.com/Beep6581/RawTherapee/pull/6434

The crop mode issue is quite a bit tricker, because it may require an overhaul of our `camconst.json` format to accommodate raw files from the same camera with different dimensions.

**Comment by rebio** [comment]
@Thanatomanic Thanks for the note.

Is there anything I can test or try for the crop issue?
I also added a comment for the issue of "Image 4" from the original list.

**Comment by rebio** [comment]
@Thanatomanic 
Continuing from #6621: 
I mixed the issues in my head.
CRAW works, 1.6 crop still gives me that black border as described in the innitial report.

I tested this image: Canon EOS R5, CRAW 1.6crop no dualpixel (a704fe3a62fbcd4aac431394e41d32307f1ba70f) from https://raw.pixls.us/
I have tested this issue on version 5.9-4-g9b79c241d (latest AppImage available for download)

**Comment by Lawrence37** [comment]
Probably #6473 plus the appropriate camconst entry will fix the black borders problem.

**Comment by Lawrence37** [comment]
@rebio #6473 should solve the black border problem for EOS R5 crop images. If you want, you can try the AppImage from [here](https://github.com/Beep6581/RawTherapee/actions/runs/3723618422) once it is built.

**Comment by Thanatomanic** [comment]
Black border issue should be fixed by https://github.com/Beep6581/RawTherapee/commit/d74524f2c57c4a4084352281ec7e6b3ba3f7e1ac. CRAW reading is still not possible, but is outside the scope of this issue.
